### ISSUE TITLE
Parameter mappings to `field-literal` don't work

### ISSUE THREAD

**Issue by tlrobinson** [metabase/metabase#11860]
As of #11827 it's possible to map parameters to `field-literal`s on nested queries, however the backend throws an error. It actually looks like there's a TODO specifically for this case: https://github.com/metabase/metabase/blob/master/src/metabase/models/params.clj#L28

```
02-05 11:23:06 WARN middleware.process-userland-query :: Query failure {:status :failed,
 :class clojure.lang.ExceptionInfo,
 :error (not (integer? "CREATED_AT")),
 :stacktrace
 ("--> models.params$eval38277$field_form__GT_id__38282.invoke(params.clj:21)"
  "query_processor.middleware.parameters.mbql$eval51604$build_filter_clause__51609$fn__51619.invoke(mbql.clj:50)"
  "query_processor.middleware.parameters.mbql$eval51604$build_filter_clause__51609.invoke(mbql.clj:39)"
  "query_processor.middleware.parameters.mbql$expand.invokeStatic(mbql.clj:78)"
  "query_processor.middleware.parameters.mbql$expand.invoke(mbql.clj:65)"
  "query_processor.middleware.parameters$expand_mbql_params.invokeStatic(parameters.clj:37)"
  "query_processor.middleware.parameters$expand_mbql_params.invoke(parameters.clj:33)"
  "query_processor.middleware.parameters$expand_one.invokeStatic(parameters.clj:49)"
  "query_processor.middleware.parameters$expand_one.invoke(parameters.clj:41)"
  "query_processor.middleware.parameters$expand_all$replace_51714__51715.invoke(parameters.clj:59)"
  "mbql.util.match$replace_in_collection$iter__31684__31688$fn__31689.invoke(match.clj:132)"
  "mbql.util.match$replace_in_collection.invokeStatic(match.clj:131)"
  "mbql.util.match$replace_in_collection.invoke(match.clj:126)"
  "query_processor.middleware.parameters$expand_all$replace_51714__51715.invoke(parameters.clj:59)"
  "query_processor.middleware.parameters$expand_all.invokeStatic(parameters.clj:59)"
  "query_processor.middleware.parameters$expand_all.invoke(parameters.clj:53)"
  "query_processor.middleware.parameters$expand_all.invokeStatic(parameters.clj:56)"
  "query_processor.middleware.parameters$expand_all.invoke(parameters.clj:53)"
  "query_processor.middleware.parameters$expand_parameters.invokeStatic(parameters.clj:77)"
  "query_processor.middleware.parameters$expand_parameters.invoke(parameters.clj:73)"
  "query_processor.middleware.parameters$eval51730$substitute_parameters_STAR___51735$fn__51736.invoke(parameters.clj:82)"
  "query_processor.middleware.parameters$eval51730$substitute_parameters_STAR___51735.invoke(parameters.clj:79)"
  "query_processor.middleware.driver_specific$process_query_in_context$fn__50540.invoke(driver_specific.clj:12)"
  "query_processor.middleware.resolve_driver$resolve_driver$fn__52119.invoke(resolve_driver.clj:22)"
  "query_processor.middleware.store$initialize_store$fn__55627$fn__55628.invoke(store.clj:11)"
  "query_processor.store$do_with_store.invokeStatic(store.clj:46)"
  "query_processor.store$do_with_store.invoke(store.clj:40)"
  "query_processor.middleware.store$initialize_store$fn__55627.invoke(store.clj:10)"
  "query_processor.middleware.async$async__GT_sync$fn__48180.invoke(async.clj:23)"
  "query_processor.middleware.async_wait$runnable$fn__49459.invoke(async_wait.clj:89)"),
 :query
 {:constraints {:max-results 10000, :max-results-bare-rows 2000},
  :type :query,
  :middleware {:userland-query? true},
  :info
  {:executed-by 1,
   :context :question,
   :card-id 1,
   :dashboard-id nil,
   :query-hash [17, -8, -125, 27, 47, -32, 107, -81, 116, 31, -45, 14, -73, 73, 106, 33, -124, 82, 32, -79, 117, 42, -107, 4, -56, 116, 25, 118, -45, -100, -95, 29]},
  :query
  {:source-query {:source-table 2, :aggregation [[:count]], :breakout [[:datetime-field [:fk-> [:field-id 11] [:field-id 7]] :month]]}, :filter [:> [:field-literal "count" :type/Integer] "100"]},
  :parameters [{:type "date/single", :target ["dimension" ["field-literal" "CREATED_AT" "type/DateTime"]], :value "2020-02-11"}],
  :async? true,
  :cache-ttl nil},
 :preprocessed nil,
 :native nil,
 :ex-data {:type :schema.core/error, :value "CREATED_AT", :error (not (integer? "CREATED_AT"))}}
```

**Comment by walterl** [comment]
This query (looks the same as the one in your error) works for me.

![image](https://user-images.githubusercontent.com/23798/76578421-2fbbca00-64d1-11ea-9f5f-84b1dd7dfcea.png)

![image](https://user-images.githubusercontent.com/23798/76578475-6560b300-64d1-11ea-9a33-727872c996d0.png)

**Comment by flamber** [comment]
If I understand this correctly, it's what is causing #11007

**Comment by flamber** [comment]
Fixed on `master` - most likely by #12531, if I understand this correctly.
